file-proxy/README.md

```diff
@@ -0,0 +1,211 @@
+# file-proxy
+
+This set of examples shows how to implement a set of webtask endpoints for file upload and download. Starting out with a basic upload/download proxy to demonstrate the pattern, this pattern can be extended to include additional features, like encryption, authentication, metadata support, multiple file upload support, and extension support.  
+Be aware that most of the examples are not production-ready with regards to things like error handling and security.
+
+## Included examples
+[simple-file-proxy](#simple-file-proxy)  
+[multiple-file-proxy](#multiple-file-proxy)  
+[metadata-file-proxy](#metadata-file-proxy)  
+[extension-file-proxy](#extension-file-proxy)  
+[encrypted-file-proxy](#encrypted-file-proxy)  
+[auth-file-proxy](#auth-file-proxy)
+
+## Getting started
+
+### Setup Graphcool project
+
+Make sure you have the Graphcool cli installed:
+
+```sh
+npm -g install graphcool
+```
+
+### Setup webtask.io
+
+Install the webtask.io cli, and initialize your webtask.io profile, by running:
+
+```sh
+npm -g install wt-cli
+wt init
+```
+
+If you have never used wt-cli before, the `wt init` command will ask for your e-mail address or phone number to set up or connect to your webtask container.
+
+## Examples
+
+### simple-file-proxy
+
+This is the starting point for the examples. It is a basic file upload/download proxy.
+
+First, initialize a new Graphcool project with the required schema, by running:
+
+```sh
+graphcool init --schema ./schemas/simple-file-proxy.graphql
+```
+
+Note the ID that this command returns. We will need this project ID later on.
+
+You can deploy the webtask for this example to webtask.io by running the following command:
+
+```sh
+wt create ./simple-file-proxy.js --name simple
+```
+
+After running this command, `wt-cli` will return the url of your webtask. Using the `-name` argument, you can specify a different name for the webtask (the last part of the url).
+
+Your webtask is now deployed! You can upload a file using any tool that support `multipart/form-data`, like Postman, or the command-line tool `curl`:
+
+```sh
+curl -X POST '<webtask endpoint url>/__PROJECT_ID__' -F "data=@small-logo.png;filename=small.png"
```

Comment:
Nit: use single quotes throughout to be consistent:
curl -X POST '<webtask endpoint url>/__GRAPHCOOL_PROJECT_ID__' -F 'data=@small-logo.png;filename=small.png'